### PR TITLE
BACKLOG-15625 issue with z-index on page composer add in return of uuid.

### PR DESCRIPTION
**Pull request by XiangYuHe**
## JIRA

https://jira.jahia.org/browse/BACKLOG-15625

## Description

Issue with "page composer" as it has display issue with z-index in comparison with jcontent.
Business requested that the "edit" onclick to open in a new tab instead.

The reason I am not using the existing code is the action is tightly bind with redux action.
Plus do not want to affect existing "edit" functionalities.

### CODE REVIEW

File: src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
+import {useSelector} from 'react-redux';
 import {isImageFile} from '../../ContentLayout.utils';
 
 const EditButton = props => {
-    const {status, file, t, path} = props;
+    const {status, file, t, uuid} = props;
 
-    if (isImageFile(file.name)) {
+    const {language} = useSelector(state => ({language: state.language}));
+    const location = (window.location.pathname.split('/').length > 1) ? `/${window.location.pathname.split('/')[1]}` : '/';
```
Comment (AKarmanov) [inline src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx]: Maybe we can extract window.location.pathname.split('/') to constant so we don't do it twice and it's a little bit shorter?

File: src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+    const url = `${location}/content-editor/${language}/edit/${uuid}`;
+
+    if (isImageFile(file.name) && uuid !== null) {
```
Comment (AKarmanov) [inline src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx:16]: Can these two ifs bee merged into one?

Comment (XiangYuHe) [reply src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx:16]: Personally I would put those differently 
As one is checking the uuid of the node it self while the other is checking if file is an image (different purpose)

Comment (AKarmanov) [reply src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx:16]: For me it's more about not having complexity of nested IF, in the end the purpose of the if check is to determine if component should be displayed so I think we would be totally ok merging the two.

Comment (XiangYuHe) [reply src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx:16]: Ah I see where you are going with that
I can combo those into an additional function that call displayEditButton

File: src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx (hunk 1 above)
Comment (XiangYuHe) [reply src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx]: Yup sounds good.

**Review by AKarmanov — APPROVED** [review]

File: src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx (hunk 2 above)
Comment (XiangYuHe) [reply src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/EditButton.jsx]: Updated to 1 if statement 👍

**Pull request merged**